X11: Can't start the editor

**Operating system or device - Godot version:**
Linux/X11, NVIDIA 364.19, Git head: 3b2c96139b39ddcfbadcd21b61e5c1ff608ad95e

**Issue description**
Can't start the editor, but can start a game.

**Steps to reproduce:**
Start the editor or project browser.

Example console output:
```bash
$ bin/godot.x11.tools.64 -path ../godot-demo-projects-master/2d/platformer -editor
LANG IS en
DPI MODE: 0
X Error of failed request:  BadRequest (invalid request code or no such operation)
  Major opcode of failed request:  140 (RANDR)
  Minor opcode of failed request:  42 ()
  Serial number of failed request:  229
  Current serial number in output stream:  229
$ echo $?
1
```
or
```bash
$ bin/godot.x11.tools.64
LANG IS en
DPI MODE: 0
X Error of failed request:  BadRequest (invalid request code or no such operation)
  Major opcode of failed request:  140 (RANDR)
  Minor opcode of failed request:  42 ()
  Serial number of failed request:  228
  Current serial number in output stream:  228
$ echo $?
1
```

Compilation command: `$ scons -j 8 platform=x11 target=debug CC='gcc-5' CXX='g++-5' tools=yes`

Tested on GCC 5.4.0 (from AUR) and Clang 3.8.0.

Did not compile:
```bash
platform/x11/os_x11.cpp: In member function 'virtual int OS_X11::get_screen_dpi(int) const':
platform/x11/os_x11.cpp:761:29: error: cannot convert 'xrr_monitor_info* {aka _xrr_monitor_info*}' to 'XRRMonitorInfo* {aka _XRRMonitorInfo*}' for argument '1' to 'void XRRFreeMonitors(XRRMonitorInfo*)'
     XRRFreeMonitors(monitors);
                             ^
platform/x11/os_x11.cpp:764:28: error: cannot convert 'xrr_monitor_info* {aka _xrr_monitor_info*}' to 'XRRMonitorInfo* {aka _XRRMonitorInfo*}' for argument '1' to 'void XRRFreeMonitors(XRRMonitorInfo*)'
    XRRFreeMonitors(monitors);
                            ^
```

So `XRRFreeMonitors()` should be also resolved `via libdl`...

Maybe move the version check code before resolving the `XRRGetMonitors` to `OS_X11::initialize`? Then it will have a `NULL`, so the version check won't be necessary later.

--- a/platform/x11/os_x11.cpp
+++ b/platform/x11/os_x11.cpp
@@ -120,15 +120,33 @@ void OS_X11::initialize(const VideoMode& p_desired,int p_video_driver,int p_audi
 
 	const char* err;
 	xrr_get_monitors = NULL;
+	xrr_free_monitors = NULL;
+	int xrandr_major = 0;
+	int xrandr_minor = 0;
+	int event_base, error_base;
+	xrandr_ext_ok = XRRQueryExtension(x11_display,&event_base, &error_base);
 	xrandr_handle = dlopen("libXrandr.so", RTLD_LAZY);
 	err = dlerror();
 	if (!xrandr_handle) {
-		fprintf(stderr, "could not load libXrandr.so, dpi detection disabled. Error: %s\n", err);
+		fprintf(stderr, "could not load libXrandr.so, Error: %s\n", err);
 	}
 	else {
-		xrr_get_monitors = (xrr_get_monitors_t) dlsym(xrandr_handle, "XRRGetMonitors");
-		if (!xrr_get_monitors)
-			fprintf(stderr, "could not find symbol XRRGetMonitors, dpi detection will only work on the first screen\nError: %s\n", err);
+		XRRQueryVersion(x11_display, &xrandr_major, &xrandr_minor);
+		if (((xrandr_major << 8) | xrandr_minor) >= 0x0105) {
+			xrr_get_monitors = (xrr_get_monitors_t) dlsym(xrandr_handle, "XRRGetMonitors");
+			if (!xrr_get_monitors) {
+				err = dlerror();
+				fprintf(stderr, "could not find symbol XRRGetMonitors\nError: %s\n", err);
+			}
+			else {
+				xrr_free_monitors = (xrr_free_monitors_t) dlsym(xrandr_handle, "XRRFreeMonitors");
+				if (!xrr_free_monitors) {
+					err = dlerror();
+					fprintf(stderr, "could not find XRRFreeMonitors\nError: %s\n", err);
+					xrr_get_monitors = NULL;
+				}
+			}
+		}
 	}
 
 	xim = XOpenIM (x11_display, NULL, NULL, NULL);
@@ -745,19 +763,18 @@ int OS_X11::get_screen_dpi(int p_screen) const {
 	ERR_FAIL_INDEX_V(p_screen, get_screen_count(), 0);
 
 	//Get physical monitor Dimensions through XRandR and calculate dpi
-	int event_base, error_base;
-	const Bool ext_okay = XRRQueryExtension(x11_display,&event_base, &error_base);
-
 	Size2 sc = get_screen_size(p_screen);
-	if (ext_okay) {
+	if (xrandr_ext_ok) {
 		int count = 0;
 		if (xrr_get_monitors) {
 			xrr_monitor_info *monitors = xrr_get_monitors(x11_display, x11_window, true, &count);
 			if (p_screen < count) {
 				double xdpi = sc.width  / (double) monitors[p_screen].mwidth  * 25.4;
 				double ydpi = sc.height / (double) monitors[p_screen].mheight * 25.4;
+				xrr_free_monitors(monitors);
 				return (xdpi + ydpi) / 2;
 			}
+			xrr_free_monitors(monitors);
 		}
 		else if (p_screen == 0) {
 			XRRScreenSize *sizes = XRRSizes(x11_display, 0, &count);

--- a/platform/x11/os_x11.h
+++ b/platform/x11/os_x11.h
@@ -178,8 +178,11 @@ class OS_X11 : public OS_Unix {
 	void set_wm_fullscreen(bool p_enabled);
 
 	typedef xrr_monitor_info* (*xrr_get_monitors_t)(Display *dpy, Window window, Bool get_active, int *nmonitors);
+	typedef void (*xrr_free_monitors_t)(xrr_monitor_info* monitors);
 	xrr_get_monitors_t xrr_get_monitors;
+	xrr_free_monitors_t xrr_free_monitors;
 	void *xrandr_handle;
+	Bool xrandr_ext_ok;
 
 protected:
 

